sycl/source/detail/config.hpp

```diff
@@ -698,6 +698,57 @@ template <> class SYCLConfig<SYCL_JIT_AMDGCN_PTX_TARGET_FEATURES> {
   }
 };
 
+// SYCL_CACHE_TRACE accepts the following values:
+// 0 - no tracing
+// 1 - trace disk cache
+// 2 - trace in-memory cache
+// 3 - trace disk and in-memory cache
+// 4 - trace kernel_compiler
+// 5 - trace disk and kernel_compiler
+// 6 - trace in-memory and kernel_compiler
+// 7 - trace disk, in-memory and kernel_compiler
+// <Any other non-null value> - trace disk cache (Legacy behavior)
+template <> class SYCLConfig<SYCL_CACHE_TRACE> {
+  using BaseT = SYCLConfigBase<SYCL_CACHE_TRACE>;
+
+public:
+  static unsigned int get() { return getCachedValue(); }
+  static void reset() { (void)getCachedValue(true); }
+  static bool isTraceDiskCache() { return getCachedValue() & 1; }
```

Comment:
we should make the bitmask be an actual enum, then & it against those names, instead of & against constant integers